Document that `Node.duplicate` also duplicates its children.

vinni-richburgh opened the pull request:
The documentation of Node.duplicate does not mention that the Node's subtree is duplicated, as well. (see https://github.com/godotengine/godot/issues/95895)

"[...] children copied from the original." implies that their copies are already attached to the new node, so I think this is really all the additional info required.

vinni-richburgh commented:
Just gave the commit a more descriptive commit message.

"Added to Node.xml file." -> "In Node.xml mentioned that Node::duplicate also duplicates its children."
https://github.com/godotengine/godot/commit/9b0c83aa8f47bcc39c7c327fac0257ef6d18191f -> https://github.com/godotengine/godot/commit/22b7672295d6f7371934172897b6b4eaf3bae9d4

Mickeon commented on doc/classes/Node.xml:
```diff
@@ -255,7 +255,7 @@
 			<param index="0" name="flags" type="int" default="15" />
 			<description>
-				Duplicates the node, returning a new node with all of its properties, signals and groups copied from the original. The behavior can be tweaked through the [param flags] (see [enum DuplicateFlags]).
+				Duplicates the node, returning a new node with all of its properties, signals, groups and children copied from the original. The behavior can be tweaked through the [param flags] (see [enum DuplicateFlags]).
```
It looks nitpicky but we actually use the [Oxford comma](https://docs.godotengine.org/en/stable/contributing/documentation/docs_writing_guidelines.html#use-the-oxford-comma-to-enumerate-anything), so we may as well fix this in one sweep, too.
```suggestion
				Duplicates the node, returning a new node with all of its properties, signals, groups, and children copied from the original. The behavior can be tweaked through the [param flags] (see [enum DuplicateFlags]).
```

Mickeon reviewed (approved):
This is fine. In hindsight I should've added it in https://github.com/godotengine/godot/pull/68560 .

It could be nice to squash these commits, see [the Pull Request Workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase).

vinni-richburgh committed:
In Node.xml mentioned that Node::duplicate also copies children.

The list now uses the Oxford comma.

vinni-richburgh commented:
I think now we are there.
Squashed the commits together and adjusted the commit message to include the addition of the Oxford comma.
If there's anything else I can do, please let me know!

AThousandShips commented:
To fix run:
```
git reset --hard 07191b4
```
And then re-do the rebsse, finally pushing with:
```
git push --force
```

vinni-richburgh commented:
Now I got it, right?
1 commit ahead of godotengine/godot master and 1 file changed.

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:

vinni-richburgh commented:
Feels great☺️
Thanks to all of you for your patience!
This surely won't be the last time, already learned so much in the process.
